minetest/minetest
# Breaking world limits. Part 1

- Goal of the PR
Start process of breaking 62000 nodes limit   
- How does the PR work?
Adding two alias types:
pos_t=s16 for nodes
bpos_t=s16 for blocks (can be changed to pos_t, but makes good visible difference between node and block coords)
adding configurable cmake variable:
USE_POS32=false
for switching pos_t=s32 , bpos_t=s32

all uses of map s16 changed to pos_t and bpos_t
now by default all types stays s16 inside - nothing broken. 

- Does it resolve any reported issue?
https://github.com/minetest/minetest/issues/8660
https://github.com/minetest/minetest/issues/10257


- If not a bug fix, why is this PR needed? What usecases does it solve?
62000^3 is too small for me.

## To do
(later in other prs)
Do something with object's v3f - its too buggy for big values.
Change them to doubles or keep v3s32 offset near pos
Fix teleport to huge coordinates

This PR is Ready for Review.


USE_POS32 broke client-server protocol, it can be enabled after fixing all bugs in some major versions. 

## How to test
cmake -DUSE_POS32=1 .

Now works with devtest game
dummy, leveldb, redis, postgres backends should work.
sqlite - sorry, need to change table.

part 1: this pr. Only s16->s32
part 2: https://github.com/minetest/minetest/pull/11910 add float -> double (smoother movement& objects)
part 3: https://github.com/minetest/minetest/pull/12142 add network compatibility with old clients
